### PR TITLE
Add support for generating chardevs and blockdevs.

### PR DESCRIPTION
**Pull request by andrewchambers**

### CODE REVIEW

File: test/0001-fspec-tar-sanity.test [hunk 1]
```diff
@@ -12,12 +12,33 @@ type=reg
...
 test -f bar
 test -d baz
+test -p fif
 
 test "0700" = "$(fmode foo)"
 test "0644" = "$(fmode bar)"
 test "0755" = "$(fmode baz)"
+test "0644" = "$(fmode fif)"
+
+cat <<EOF | fspec-tar | tar -tf - > out.list
```
Comment (michaelforney) [inline test/0001-fspec-tar-sanity.test]: `cat` is unnecessary here. You can just use `fspec-tar <<EOF`.

**Review by michaelforney — APPROVED** [review]

**Comment by andrewchambers** [comment]
@michaelforney I removed the ugly fmode instead opting for self testing. The current test failure is a bit confusing to me.

**Comment by michaelforney** [comment]
Looks like you want `archive_entry_set_rdev` and `archive_entry_rdev` instead of the _dev functions.

**Comment by andrewchambers** [comment]
Hmm no luck with rdev.

**Comment by michaelforney** [comment]
Hmm... are you sure. It worked for me on 0001-fspec-tar-sanity.test.

0002-fspec-fromdir-sanity still fails though. The tar command is not correct. It should be

```
-tar -C foo -f . | fspec-fromtar > b.fspec
+tar -C foo -cf - a.txt b.sh | fspec-fromtar > b.fspec
```

However, this doesn't work since the tar archive has my own UID/GID.

Why don't we diff a.fspec against our expectation?

**Comment by andrewchambers** [comment]
Ah, we can, I think I didn't because I was worried about umask breaking tests. We can just chmod though.

**Pull request merged**